web/src/features/map/Map.tsx

```diff
@@ -153,14 +156,46 @@ export default function MapPage({ onMapLoad }: MapPageProps): ReactElement {
 
   useEffect(() => {
     // Run on first load to center the map on the user's location
-    if (!map || isError || !isFirstLoad) {
+    if (!map || isError || !isFirstLoad || !isSourceLoaded || !callerLocation) {
       return;
     }
-    if (data?.callerLocation && !selectedZoneId) {
-      map.flyTo({ center: [data.callerLocation[0], data.callerLocation[1]] });
+    if (callerLocation && !selectedZoneId) {
```

Comment:
The check for `callerLocation` here seems unnecessary as we're returning above if it does not exist